arangod/IResearch/IResearchViewCoordinator.cpp

```diff
@@ -67,7 +69,45 @@ arangodb::Result IResearchViewCoordinator::appendVelocyPackDetailed(
     );
   }
 
+  // open up a read transaction and add all linked collections to verify that
+  // the current user has access
+
   arangodb::velocypack::Builder links;
+  std::vector<std::string> collections;
+  for (auto& entry: _collections) {
+    collections.emplace_back(entry.second.first);
+  }
+
+  static std::vector<std::string> const EMPTY;
+  // use default lock timeout
+  arangodb::transaction::Options options;
+  options.waitForSync = false;
+  options.allowImplicitCollections = false;
+
+  Result result = arangodb::basics::catchToResult([&]() -> arangodb::Result {
+    arangodb::transaction::Methods trx(
+      transaction::StandaloneContext::Create(vocbase()),
```

Comment:
let's avoid having heap allocation there, better to use `shared_ptr` aliasing constructor